dbms/src/Interpreters/InterpreterSystemQuery.cpp

```diff
@@ -338,7 +339,17 @@ void InterpreterSystemQuery::syncReplica(ASTSystemQuery & query)
     StoragePtr table = context.getTable(database_name, table_name);
 
     if (auto storage_replicated = dynamic_cast<StorageReplicatedMergeTree *>(table.get()))
-        storage_replicated->waitForShrinkingQueueSize(0, context.getSettingsRef().receive_timeout.value.milliseconds());
```

Comment:
Note the use of `receive_timeout.value.milliseconds()` here instead of `totalMilliseconds()`!!!